client/components/LearnerViewDirective.js

```diff
@@ -477,10 +477,22 @@ tie.directive('learnerView', [function() {
             feedbackDiv.scrollTop = feedbackDiv.scrollHeight +
               additionalHeightForLoadingIndicator;
             $timeout(function() {
-              SolutionHandlerService.processSolutionAsync(
-                tasks[currentTaskIndex], code,
+              var index = 0;
+              var fn = function() {
+                SolutionHandlerService.processSolutionAsync(
+                tasks[index], code,
                 question.getAuxiliaryCode(language), language
-                ).then(setFeedback);
+                ).then(function(feedback) {
+                  if (index === currentTaskIndex ||
+                      !feedback.isAnswerCorrect()) {
+                    setFeedback(feedback);
```

Comment:
The functionality of providing feedback to an earlier part of the question is new, so I wanted to check whether there's the potential for confusion if the user is solving task 3 and gets feedback for task 2. Can you post some screenshots showing what such a flow would look like?